sandbox/modules/identity/src/main/java/org/opensearch/identity/authz/OpenSearchPermission.java

```diff
@@ -68,9 +123,17 @@ public boolean implies(Permission p) {
         // For all requested index patterns (ip):
         // ip must match at least one pattern that the user has been granted access to
 
-        boolean allRequestedPatternsMatchGranted = requestedPermission.indexPatterns.stream()
-            .allMatch(requstedIp -> this.indexPatterns.stream().anyMatch(ip -> Glob.globMatch(ip, requstedIp)));
+        boolean allRequestedPatternsMatchGranted = requestedPermission.resourcePatterns.stream()
+            .allMatch(requstedIp -> this.resourcePatterns.stream().anyMatch(ip -> Glob.globMatch(ip, requstedIp)));
 
         return allRequestedPatternsMatchGranted;
     }
+
+    public boolean equals(OpenSearchPermission secondPermission) {
```

Comment:
Please see the example in StringPrincipal for overriding an object's equal method. 

It may make sense to break this class up into ClusterPermission and IndexPermission. A ClusterPermission equals another ClusterPermission when the action names are similar. This will be different for IndexPermissions based on how we model an IndexPermission.